lib/matplotlib/colors.py

```diff
@@ -1885,12 +1885,16 @@ def __call__(self, value, clip=None):
                 result = np.ma.array(np.clip(result.filled(vmax), vmin, vmax),
                                      mask=mask)
             resdat = result.data
-            resdat -= vmin
-            resdat[resdat < 0] = 0
-            np.power(resdat, gamma, resdat)
-            resdat /= (vmax - vmin) ** gamma
+            # These values are masked later, but set to zero now to prevent
+            # invalid value warnings when taking the power
+            under_mask = resdat < vmin
+            resdat[under_mask] = vmin
 
-            result = np.ma.array(resdat, mask=result.mask, copy=False)
+            resdat = (resdat - vmin) / (vmax - vmin)
+            resdat = resdat ** gamma
+
+            mask = result.mask | under_mask
+            result = np.ma.array(resdat, mask=mask, copy=False)
```

Comment:
I think this should be 
```python
good = resdat >= vmin
resdat = (resdat - vmin) / (vmax - vmin)
resdat[good] = resdat[good]**gamma
```
This will make the negative part of the scale linear, but it's better than setting it to be masked.

Reply:
Why do we want to make the negative part of the scale linear? In the same way a log scale is valid for `x > 0`, and masks the output where any values have `x <= 0`, the `PowerNorm` as implemented is valid for `x >= vmin`, so it is consistent to mask for `x < vmin`.

Reply:
Because the way you are doing it here, x < vmin is always masked.  

For LogNorm x<vmin is _only_ masked if x<0.  However, 0<x<vmin is _not_ masked, but set to a number less than 0, and hence shows up as the under-color in the colormapping.  

This proposal will allow x<vmin, and always map it to a number less than zero, and hence show up as the under-color in the color mapping.     

Again, the reason is that the order of operations is different for PowerNorm.  For LogNorm we do

```
norm = (log10(x) - log10(vmin))/ (log10(vmax)-log10(vmin)
```

for PowerNorm we do

```
norm = ((x-vmin)/(vmax-vmin))**gamma
```

so whether we apply the gamma outside the range [vmin, vmax] doesn't matter too much.